types/types.go

```diff
@@ -451,14 +457,18 @@ type SystemContext struct {
 	RegistriesDirPath string
 	// Path to the system-wide registries configuration file
 	SystemRegistriesConfPath string
-	// If not "", overrides the default path for the authentication file
+	// If not "", overrides the default path for the authentication file.  Ignored for reading if AuthConfig is not "".
+	//
+	// Deprecated: Use the GetAuth property and github.com/containers/image/pkg/docker/config's AuthStore instead.
```

Comment:
It’s not obviously attractive to force the users to create extra objects when they could just set a clearly-named field. Sure, the consumer of the data might need to define what happens if multiple fields are set, or refuse such configurations, but that seems worth it.

Reply:
> Sure, the consumer of the data might need to define what happens if multiple fields are set, or refuse such configurations, but that seems worth it.

`SystemContext` seems complicated enough already, to me it is a clear win to factor out the auth-configuration properties into other locations.  For folks who currently use `SystemContext`, the transitions seem pretty clear to me, although I can work up migration docs if that would help address your concerns (or even just give us something more concrete to anchor this discussion).